src/openloco/openloco.cpp

```diff
@@ -308,6 +316,126 @@ namespace openloco
         gfx::clear(gfx::screen_dpi(), 0x0A0A0A0A);
     }
 
+    // 0x00428E47
+    static void sub_428E47()
+    {
+        call(0x00428E47);
+    }
+
+    // 0x004C96E7
+    static void handle_input()
+    {
+        call(0x004C96E7);
+    }
+
+    // 0x004383ED
+    static void sub_4383ED()
+    {
+        call(0x004383ED);
+    }
+
+    // 0x0046E388
+    static void sub_46E388()
+    {
+        call(0x0046E388);
+    }
+
+    // 0x004317BD
+    static uint32_t sub_4317BD()
+    {
+        registers regs;
+        call(0x004317BD, regs);
+
+        return regs.eax;
+    }
+
+    // 0x0046E4E3
+    static void sub_46E4E3()
+    {
+        call(0x0046E4E3);
+    }
+
+    void sub_431695()
+    {
+        if (!isMultiplayer())
+        {
+            _updating_company_id = companymgr::get_controlling_id();
+            for (auto i = 0; i < _F253A0; i++)
+            {
+                sub_428E47();
+                WindowManager::dispatchUpdateAll();
+            }
+
+            input::process_keyboard_input();
+            WindowManager::update();
+            handle_input();
+            sub_4383ED();
+            return;
+        }
+
+        // Only run every other tick?
+        if (_525F62 % 2 != 0)
+        {
+            return;
+        }
+
+        // Host/client?
+        if ((get_screen_flags() & screen_flags::unknown_3) != 0)
```

Comment:
I think the branches have been implemented incorrectly per `4316EE`. They should either be interchanged, or this condition should be `== 0`.

Reply:
![image](https://user-images.githubusercontent.com/737603/53291021-f1137e80-37ac-11e9-84d6-d24005243e9f.png)

Looks to me like the srand_0 stuff is only happening when `(flags & 8) == 0`

Reply:
You're right. Carry on then.